core/src/drop_bank_service.rs

```diff
@@ -0,0 +1,38 @@
+use solana_measure::measure::Measure;
+use solana_runtime::bank::Bank;
+use std::{
+    sync::{mpsc::Receiver, Arc},
+    thread::{self, Builder, JoinHandle},
+};
+
+pub struct DropBankService {
+    thread_hdl: JoinHandle<()>,
+}
+
+impl DropBankService {
+    pub fn new(bank_receiver: Receiver<Vec<Arc<Bank>>>) -> Self {
+        let thread_hdl = Builder::new()
+            .name("sol-drop-b-service".to_string())
+            .spawn(move || {
+                for banks in bank_receiver.iter() {
+                    let len = banks.len();
+                    let mut dropped_banks_time = Measure::start("drop_banks");
+                    drop(banks);
+                    dropped_banks_time.stop();
+                    if dropped_banks_time.as_ms() > 10 {
```

Comment:
Maybe worth logging a datapoint for each individual bank for each slot in case it's something bank specific?

Also not a big deal, but I think it's possible some other thread has an outstanding reference to some `Arc` root bank even after it was removed in the replay path (something like, RPC or the vote listener) but this at least unblocks replay.